src/super_gradients/training/datasets/detection_datasets/detection_dataset.py

```diff
@@ -529,34 +539,62 @@ def plot(self, max_samples_per_plot: int = 16, n_plots: int = 1, plot_transforme
         input_format = self.output_target_format if plot_transformed_data else self.original_target_format
         if isinstance(input_format, DetectionTargetsFormat):
             raise ValueError(
-                "Plot is not supported for DetectionTargetsFormat. Please set original_target_format to be an isntance of ConcatenateTransform instead."
+                "Plot is not supported for DetectionTargetsFormat. Please set original_target_format to be an instance of ConcatenateTransform instead."
             )
-        target_format_transform = DetectionTargetsFormatTransform(input_format=input_format, output_format=XYXY_LABEL)
 
         for plot_i in range(n_plots):
+
             fig = plt.figure(figsize=(10, 10))
+
             n_subplot = int(np.ceil(max_samples_per_plot**0.5))
+
+            # Plot `max_samples_per_plot` images.
             for img_i in range(max_samples_per_plot):
                 index = img_i + plot_i * 16
 
+                # LOAD IMAGE/TARGETS
                 if plot_transformed_data:
+                    # Access to the image and the target AFTER self.transform
                     image, targets, *_ = self[img_i + plot_i * 16]
-                    if image.shape[0] == 3:
-                        image = image.transpose(1, 2, 0).astype(np.int32)
                 else:
+                    # Access to the image and the target BEFORE self.transform
                     sample = self.get_sample(index=index, ignore_empty_annotations=self.ignore_empty_annotations)
                     image, targets = sample["image"], sample["target"]
 
-                sample = target_format_transform({"image": image, "target": targets})
+                # FORMAT TARGETS
+                if image.shape[0] in (1, 3):  # (C, H, W) -> (H, W, C)
+                    image = image.transpose((1, 2, 0))
 
-                # shape = [padding_size x 4] (The dataset will most likely pad the targets to a fixed dim)
-                boxes = sample["target"][:, 0:4]
+                image = image.astype(np.uint8)
+                image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)  # Detection dataset works with BGR images, so we have to convert to RGB
+
+                # Convert to XYXY_LABEL format
+                targets_format_converter = ConcatenatedTensorFormatConverter(input_format=input_format, output_format=XYXY_LABEL, image_shape=image.shape)
+                targets_xyxy_label = targets_format_converter(targets)
+
+                # PLOT TARGETS
+                color_mapping = generate_color_mapping(len(self.classes))
```

Comment:
We have DetectionVisualization that contains many similar routines we use here and I wonder maybe we can concentrate all drawing logic in that class and call DetectionVisualization._visualize_image (make sense to make method public)? WDYT?

Reply:
I pushed it here [2b2dbc7](https://github.com/Deci-AI/super-gradients/pull/1686/commits/2b2dbc784016fb255e8c9bf602b8c3fe43e99318)